runtime/compiler/control/CompilationThread.cpp

```diff
@@ -12494,3 +12530,24 @@ TR::CompilationInfo::addRemoteMethodToBeCompiled(JITaaS::J9ServerStream *stream)
       }
    return entry;
    }
+
+void
+TR::CompilationInfo::requeueOutOfProcessEntry(TR_MethodToBeCompiled *entry)
+   {
+   bool enableJITaaSPerCompConn = feGetEnv("TR_EnableJITaaSPerCompConn") ? true : false;
```

Comment:
Maybe move the ASSERT up front and execute `IProfiler_persistIprofileInfo` without any test (it's common on both paths). Then add the test for `enableJITaaSPerCompConn`